### ISSUE TITLE
Discussion: auto-inc lock deadlocks?

### ISSUE THREAD

**Issue by tiegz** [soundcloud/lhm#27]
Curious if anyone else has experienced a deadlock during an LHM migration where, for instance, on a busy table Transaction1 (replace into...) is waiting for Transaction2 (insert into...) with `lock mode AUTO-INC` and the transactions get rolled back?

In the past we've seen this once, and the most obvious answer seems that it's just an auto-increment lock deadlock. However, for most cases you're copying over the `id` field from table-to-table, which is typically an auto-increment field.

To fix this one could remove the auto-increment from that field before the LHM, and add it back to the new table's field right before the table switch. Another option would be to set [innodb_autoinc_lock_mode = 2](http://dev.mysql.com/doc/refman/5.1/en/innodb-auto-increment-handling.html#innodb-auto-increment-configurable) in mysql temporarily until the table switch.

My question is, would this be a good configurable option for LHM, since it applies to a typical database setup?

**Comment by arthurnn** [comment]
I beleive this is fixed on master, as we now have a code to retry if deadlocks are found #93 .

**Comment by meunierd** [comment]
@arthurnn I don't think that PR actually addresses the deadlock issue. That only affects the switcher and the LHMs discussed in this issue fail well before the table switch. Could this be re-opened?

**Comment by avit** [comment]
See also: #107?
